oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/OidcAuthorizationCodeAuthenticationProvider.java

```diff
@@ -154,6 +162,19 @@ public Authentication authenticate(Authentication authentication) throws Authent
 		}
 		OidcIdToken idToken = createOidcToken(clientRegistration, accessTokenResponse);
 
+		String nonceHash;
+		try {
+			nonceHash = createHash(authorizationRequest.getAttribute(IdTokenClaimNames.NONCE));
+		} catch (NoSuchAlgorithmException e) {
+			OAuth2Error nonceVerificationError = new OAuth2Error(UNVERIFIABLE_NONCE, e.getMessage(), null);
```

Comment:
We should not give too much information to the caller letting them know that the `nonce` could not be verified. A generic message is sufficient so let's reuse `INVALID_NONCE` and remove `UNVERIFIABLE_NONCE `